### PR TITLE
Quiet runner

### PR DESCRIPTION
**Pull request by cdent**
Add a -q to gabbi-run and then fix the tests to deal with urllib3 breaking stuff (see the commit messages).

### CODE REVIEW

File: gabbi/runner.py [hunk 1]
```diff
@@ -136,8 +139,14 @@ def run_suite(handle, handler_objects, host, port, prefix, force_ssl=False,
         loader, name, data, data_dir, host, port, None, None, prefix=prefix,
         handlers=handler_objects, test_loader_name='gabbi-runner')
 
+    # The default runner stream is stderr.
+    stream = sys.stderr
+    if quiet:
+        # We want to swallow the output that the runner is
+        # producing.
```
Comment (FND) [inline gabbi/runner.py]: a) weird break but okay

b) your continued resistance to ternaries is kinda hilarious

File: gabbi/runner.py [hunk 2]
```diff
@@ -213,6 +222,11 @@ def _make_argparser():
         action='store_true',
         help='Exit on first failure'
     )
+    parser.add_argument(
+        '-q', '--quiet',
+        action='store_true',
+        help='Produce no test runner output'
```
Comment (FND) [inline gabbi/runner.py]: That phrasing seems a little awkward; "Suppress test runner output" seems more straightforward?

File: gabbi/tests/test_runner.py [hunk 3]
```diff
@@ -352,6 +352,31 @@ def test_verbosity_arg_all(self):
         self.assertIn('{\n  "cat": "poppy"\n}', output)
         self.assertIn('application/json', output)
 
+    def test_quiet_is_quiet(self):
```
Comment (FND) [inline gabbi/tests/test_runner.py]: 🍪

File: gabbi/runner.py (hunk 1 above)
Comment (cdent) [reply gabbi/runner.py]: > b) your continued resistance to ternaries is kinda hilarious

I had a ternary there but changed it when I wanted the comments to be separate.

**Pull request merged**